src/glib/glib_dispatch_pool.c

```diff
@@ -0,0 +1,181 @@
+// Copyright 2015-2024 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "glibp.h"
+#include "../sub.h"
+
+static inline void
+_destroyDispatcher(natsDispatcher *d)
+{
+    if ((d == NULL) || !d->running)
+        return;
+
+    natsThread_Destroy(d->thread);
+    nats_destroyQueuedMessages(&d->queue); // there's NEVER anything there, remove?
+    natsCondition_Destroy(d->cond);
+    natsMutex_Destroy(d->mu);
+    memset(d, 0, sizeof(*d));
+}
+
+static inline natsStatus
+_startDispatcher(natsDispatcher *d, void (*threadf)(void *))
+{
+    natsStatus s = NATS_OK;
+
+    if (d->running)
+        return NATS_OK;
+
+    s = natsMutex_Create(&d->mu);
+    if (s != NATS_OK)
+        return s;
```

Comment:
Is it intentional that we are not using `NATS_UPDATE_ERR_STACK` in this file? (I forgot, but maybe we can't/should not if it is deep down at library level).

Reply:
No it is not intentional. TBH I did not yet have a chance to understand what is the idiomatic use of the macro, i.e. when is it appropriate, and when not? Should it be always do it with the exceptions of small or ...Impl style functions? Or would that tend to to over-run the error stack? I'll try to figure it out soon, and meanwhile please do point out any under/mis-use of it. Much appreciated!

Reply:
Imagine that we have these:
```
natsStatus baz(void) {
    return nats_setError(NATS_ERR, "%s", "Some error");
}
natsStatus bar(void) {
   return baz();
}
natsStatus foo(void) {
   return bar();
}
natsStatus natsConnection_Connect(...
..
s = foo();
if (s != NATS_OK)
   return s;
```
This would produce this stack:
```
Error: 1 - Error - (conn.c:2029): Some error
Stack: (library version: 3.9.0-beta)
  01 - baz
```
But if you had this instead:
```
natsStatus baz(void) {
    return nats_setError(NATS_ERR, "%s", "Some error");
}
natsStatus bar(void) {
    natsStatus s = baz();
    return NATS_UPDATE_ERR_STACK(s);
}
natsStatus foo(void) {
    natsStatus s = bar();
    return NATS_UPDATE_ERR_STACK(s);
}
natsStatus natsConnection_Connect(...
..
s = foo();
if (s != NATS_OK)
    return NATS_UPDATE_ERR_STACK(s);
```
That would produce this stack:
```
Error: 1 - Error - (conn.c:2029): Some error
Stack: (library version: 3.9.0-beta)
  01 - baz
  02 - bar
  03 - foo
  04 - natsConnection_Connect
```
So I would put it almost everywhere in case I want to know how deep the error originate from. There are cases where it is not needed because a function may return an error but the caller will always override the error returned (usually some utility or low level functions) with some more telling error.